src/pages/vagrant/docs/[[...page]].tsx

```diff
@@ -0,0 +1,53 @@
+import { ReactElement } from 'react'
+import vagrantData from 'data/vagrant.json'
+import { Product } from 'types/products'
+import SidebarSidecarLayout from 'layouts/sidebar-sidecar'
+import DocsView from 'views/docs-view'
+import Button from '@hashicorp/react-button'
+// imports below are used on server
+import { getStaticGenerationFunctions } from 'layouts/sidebar-sidecar/server'
+import fetchFileString from 'lib/fetch-file-string'
+import { getLatestVersionFromVersions } from 'lib/fetch-release-data'
+
+const basePath = 'docs'
+const baseName = 'Docs'
+const product = vagrantData as Product
+const additionalComponents = { Button }
+
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+const VagrantDocsPage = ({ mdxSource }): ReactElement => {
+  return <DocsView {...mdxSource} additionalComponents={additionalComponents} />
+}
+
+// Note that we require VMWARE_UTILITY_VERSION to be in { scope } for the MDX
+// on the  /vagrant/docs/providers/vmware/vagrant-vmware-utility page.
+// We fetch VMWARE_UTILITY_VERSION from the releases.hashicorp.com API.
+type MdxScope = { VMWARE_UTILITY_VERSION: string }
+const {
+  getStaticPaths,
+  getStaticProps,
+} = getStaticGenerationFunctions<MdxScope>({
+  product,
+  basePath,
+  baseName,
+  getScope: async () => ({
+    VMWARE_UTILITY_VERSION: await getLatestVagrantVmwareVersion(),
+  }),
+})
+
+/**
+ * Fetch the latest version of the Vagrant VMWare utility
+ * from the releases.hashicorp.com API.
+ *
+ * @returns {string} A semver string representing the latest version number
+ */
+async function getLatestVagrantVmwareVersion(): Promise<string> {
+  const url = 'https://releases.hashicorp.com/vagrant-vmware-utility/index.json'
+  const releaseData = JSON.parse(await fetchFileString(url))
```

Comment:
I think we want to use `fetchWithRetry` (in the same file as `getLatestVersionFromVersions`) here? Looks like it helps handle a race condition and I also noticed that where it's used in `src/lib/fetch-release-data.ts` sends these options with the `fetch`:

```
{
  headers: {
    'Cache-Control': 'no-cache',
  },
}
```

Reply:
✅ Have updated this - realized this *may* not be as much of an issue since we're no longer fetching a specific version? I think the issue originally came up because we were reading the "latest" version from `data/version.json`, when that latest version might not yet have been on the registry. Either way though I suppose `fetchWithRetry` can't hurt 👍